rasa/telemetry.py

```diff
@@ -484,6 +509,56 @@ def toggle_telemetry_reporting(is_enabled: bool) -> None:
     rasa_utils.write_global_config_value(CONFIG_FILE_TELEMETRY_KEY, configuration)
 
 
+@ensure_telemetry_enabled
+def initialize_error_reporting() -> None:
+    """Sets up automated error reporting.
+
+    Exceptions are reported to sentry. We avoid sending any metadata (local
+    variables, paths, ...) to make sure we don't compromise any data. Only the
+    exception and its stacktrace is logged and only if the exception origins
+    from the `rasa` package. """
+    import sentry_sdk
+    from sentry_sdk.integrations.atexit import AtexitIntegration
+    from sentry_sdk.integrations.dedupe import DedupeIntegration
+    from sentry_sdk.integrations.excepthook import ExcepthookIntegration
+
+    # key for local testing can be found at
+    # https://sentry.io/settings/rasahq/projects/rasa-open-source/install/python/
+    # for local testing, set the key using `RASA_EXCEPTION_WRITE_KEY=key rasa <command>`
+    key = sentry_write_key()
+
+    if not key:
+        return
+
+    def strip_sensitive_data(event, hint):
+        # removes any paths from stack traces (avoids e.g. sending
+        # a users home directory name if package is installed there)
+        for value in event.get("exception", {}).get("values", []):
+            for frame in value.get("stacktrace", {}).get("frames", []):
+                frame["abs_path"] = ""
+        return event
+
+    # this is a very defensive configuration, avoiding as many integrations as
+    # possible. it also submits very little data (exception with error message
+    # and line numbers).
+    sentry_sdk.init(
+        f"https://{key}.ingest.sentry.io/2801673",
+        before_send=strip_sensitive_data,
+        integrations=[
+            ExcepthookIntegration(),
+            DedupeIntegration(),
+            AtexitIntegration(lambda _, __: None),
+        ],
+        send_default_pii=False,  # activate PII filter
+        server_name=get_telemetry_id() or "UNKNOWN",
+        in_app_include=["rasa"],  # only submit errors in this package
```

Comment:
I'm not sure about this one. E.g. if there is an error with the lock store then it would be great to capture the errors directly from the redis library.

Reply:
I'd rather start with a narrow definition of this and then if we think this works for us extend this definition

Reply:
in general, I agree, we can be more open here but for the moment I think it is fine to start just with exceptions originating in our codebase.